zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -4796,6 +4837,83 @@ components:
         incidentKey:
           description: Incident key associated with this flow node instance.
           type: string
+    AdHocSubprocessActivitySearchQuery:
+      description: Ad-hoc subprocess activities search request.
+      type: object
+      required:
+        - filter
+      properties:
+        filter:
+          description: The ad-hoc subprocess activity search filters.
+          allOf:
+            - $ref: "#/components/schemas/AdHocSubprocessActivityFilter"
+    AdHocSubprocessActivityFilter:
+      description: Flow node instance filter.
+      type: object
+      required:
+        - processDefinitionKey
+        - adHocSubprocessId
+      properties:
+        processDefinitionKey:
+          description: The process definition key.
+          type: string
```

Comment:
I wasn't completely sure how to handle this as in the code we need to do some casting between string and long on this property. Kept it as string here to be consistent with other APIs and if I remember correctly the plan is to treat all keys as strings in the future?

Reply:
Yep, all keys are supposed to be string values 👍 The other controllers show to handle it, I believe.